aries_cloudagent/connections/models/diddoc_v2/tests/test_diddoc.py

```diff
@@ -598,6 +598,138 @@ async def test_universal_resolver(self):
         result3 = DIDDoc.deserialize(universal_resolver_DID_3)
         assert result3.service[0].id
 
+        # Same key reference in multiple fields
+        universal_resolver_DID_4 = {
+            "@context": "https://w3id.org/did/v0.11",
+            "id": "did:web:did.actor:alice",
+            "publicKey": [
+                {
+                    "id": "did:web:did.actor:alice#z6MkrmNwty5ajKtFqc1U48oL2MMLjWjartwc5sf2AihZwXDN",
+                    "controller": "did:web:did.actor:alice",
+                    "type": "Ed25519VerificationKey2018",
+                    "publicKeyBase58": "DK7uJiq9PnPnj7AmNZqVBFoLuwTjT1hFPrk6LSjZ2JRz",
+                }
+            ],
+            "authentication": [
+                "did:web:did.actor:alice#z6MkrmNwty5ajKtFqc1U48oL2MMLjWjartwc5sf2AihZwXDN"
+            ],
+            "assertionMethod": [
+                "did:web:did.actor:alice#z6MkrmNwty5ajKtFqc1U48oL2MMLjWjartwc5sf2AihZwXDN"
+            ],
+            "capabilityDelegation": [
+                "did:web:did.actor:alice#z6MkrmNwty5ajKtFqc1U48oL2MMLjWjartwc5sf2AihZwXDN"
+            ],
+            "capabilityInvocation": [
+                "did:web:did.actor:alice#z6MkrmNwty5ajKtFqc1U48oL2MMLjWjartwc5sf2AihZwXDN"
+            ],
+            "keyAgreement": [
+                {
+                    "id": "did:web:did.actor:alice#zC8GybikEfyNaausDA4mkT4egP7SNLx2T1d1kujLQbcP6h",
+                    "type": "X25519KeyAgreementKey2019",
+                    "controller": "did:web:did.actor:alice",
+                    "publicKeyBase58": "CaSHXEvLKS6SfN9aBfkVGBpp15jSnaHazqHgLHp8KZ3Y",
+                }
+            ],
+        }
+        result4 = DIDDoc.deserialize(universal_resolver_DID_4)
+        assert result4.authentication == result4.capability_delegation
+        assert result4.authentication == result4.capability_invocation
+        assert result4.authentication == result4.public_key
+        assert result4.authentication[0].id != result4.key_agreement[0].id
```

Comment:
We should also validate that re-serializing the document is equal to the original.

Reply:
In this cases they are not equal because the DIDDoc is "normalized" to not raise exceptions on validation/instantation phase. However I will do more specific assertions.

Reply:
Ah, I see, good point. Thanks for the clarification!